src/contract.rs

```diff
@@ -133,12 +134,15 @@ impl<'a> AffiliateSwap<'a> {
             token_in: Some(
                 Coin {
                     denom: coin.denom.clone(),
+                    // (boss): [question] not using checked_sub here as error is unexpected?
```

Comment:
Yeah I guess this is more of a style matter, rather than anything serious. IMO using checked consistently reduce decision to make. Another reason is wasm size https://rustwasm.github.io/docs/book/reference/code-size.html#avoid-panicking

Reply:
it would be nice to have a `CheckedUint128` that overloads the operators with the checked versions. That way we could do `((a+b)? * c)?`